src/main/java/jetbrains/buildServer/iaa/utils/AssignerArtifactDao.java

```diff
@@ -0,0 +1,98 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package jetbrains.buildServer.iaa.utils;
+
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import java.io.File;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.util.List;
+import jetbrains.buildServer.iaa.common.Constants;
+import jetbrains.buildServer.iaa.common.Responsibility;
+import jetbrains.buildServer.serverSide.STestRun;
+import jetbrains.buildServer.users.User;
+import jetbrains.buildServer.users.UserModelEx;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+public class AssignerArtifactDao {
+  private UserModelEx myUserModel;
+
+  public AssignerArtifactDao(@NotNull final UserModelEx userModel) {
+    myUserModel = userModel;
+  }
+
+  public void put(STestRun testRun, Responsibility responsibility) {
+    File resultsFile = this.getAssignerResultFile(testRun);
+
+    Gson gson = new GsonBuilder().registerTypeAdapter(Responsibility.class, new ResponsibilitySerializer()).create();
+    try {
+      Files.write(resultsFile.toPath(), gson.toJson(responsibility).getBytes(StandardCharsets.UTF_8));
+    } catch (IOException ex) {
+      throw new RuntimeException("An error occurs during creation of file with results");
+    }
+  }
+
+  @Nullable
+  public Responsibility get(STestRun testRun) {
+    File resultsFile = this.getAssignerResultFile(testRun);
+    if (!resultsFile.exists()) {
+      return null;
+    }
+
+    Gson gson = new GsonBuilder().registerTypeAdapter(Responsibility.class, new ResponsibilitySerializer()).create();
+    Responsibility result;
+    try {
+      List<String> responsibilityJson = Files.readAllLines(resultsFile.toPath(), StandardCharsets.UTF_8);
+      ResponsibilityPair pair = gson.fromJson(String.join("\n", responsibilityJson), ResponsibilityPair.class);
+      if (pair.investigator == null) {
+        throw new RuntimeException("Investigator is not specified!");
+      }
+      User user = myUserModel.findUserAccount(null, pair.investigator);
+      result = new Responsibility(user, pair.description);
+    } catch (IOException ex) {
+      throw new RuntimeException("An error occurs during creation of file with results");
+    }
+    return result;
+  }
+
+  private File getAssignerResultFile(STestRun testRun) {
+    File artifactDirectory = testRun.getBuild().getArtifactsDirectory();
```

Comment:
There could be used `#getArtifactsDirectory().toPath` and then use `java.nio.Path` to resolve paths.